modules/core/shared/src/main/scala/net/MessageSocket.scala

```diff
@@ -54,9 +52,9 @@ object MessageSocket {
         * total including self but not including the tag) in network order.
         */
         val receiveImpl: F[BackendMessage] = {
-          val header = (byte.asDecoder, int32.asDecoder).tupled
           bvs.read(5).flatMap { bits =>
-            val (tag, len) = header.decodeValue(bits).require
+            val tag = bits.take(8).toByte()
```

Comment:
Are you just inlining the header decoder?  Or why this change?

Reply:
Avoids allocations of a bunch of intermediate objects. Makes a small bit consistent improvement on the benchmark.